src/Orleans.Runtime/Lifecycle/IStartupTask.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+
+namespace Orleans.Runtime
+{
+    /// <summary>
+    /// Defines an action to be taken during silo startup.
+    /// </summary>
+    public interface IStartupTask
+    {
+        /// <summary>
+        /// Called after the silo has started.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token which is canceled when the method must abort.</param>
+        /// <returns>A <see cref="Task"/> representing the work performed.</returns>
+        Task OnStarted(CancellationToken cancellationToken);
```

Comment:
Is this a suitable name? Is the class suitably named? I'm open to suggestions.

Reply:
Maybe `Run` or `Execute`? It's not like this code signs up for a notification about silo startup, more like it simply wants to get executed when the environment is ready.